zeppelin-server/src/main/java/org/apache/zeppelin/rest/NotebookRestApi.java

```diff
@@ -345,16 +370,20 @@ public Response createNote(String message) throws IOException {
       }
     }
     note.addNewParagraph(subject); // add one paragraph to the last
-    String noteName = request.getName();
-    if (noteName.isEmpty()) {
-      noteName = "Note " + note.getId();
+
+    //deal with possible empty note name
+    String noteName = "Note " + note.getId();
+    if (request != null && request.getName() != null) {
+      if (!request.getName().isEmpty()) {
+        noteName = request.getName();
+      }
     }
 
     note.setName(noteName);
     note.persist(subject);
     notebookServer.broadcastNote(note);
     notebookServer.broadcastNoteList(subject, SecurityUtils.getRoles());
-    return new JsonResponse<>(Status.OK, "", note.getId()).build();
+    return new JsonResponse<>(Status.CREATED, "", note.getId()).build();
```

Comment:
I think we need to be super careful with changing an existing response code for an existing API - there is likely existing usage that could be broken with a simple code change.

I get that 201 might be better and/or more consistent with other APIs though. Do you think it makes a difference if we change it now, or we leave it as before?